core/consensus_test.go

```diff
@@ -121,6 +121,23 @@ func quorum(numNodes uint64) uint64 {
 	}
 }
 
+func commonHasQuorumFn(numNodes uint64) func(view *proto.View, messages []*proto.Message) bool {
+	return func(view *proto.View, messages []*proto.Message) bool {
+		if len(messages) > 0 {
```

Comment:
```
		if len(messages) == 0 {
			return false
		}

		switch messages[0].GetType() {
			case proto.MessageType_PREPREPARE:
				return len(messages) > 1
			case proto.MessageType_PREPARE:
				return len(messages) >= int(numNodes)-1
			case proto.MessageType_ROUND_CHANGE, proto.MessageType_COMMIT:
				return len(messages) >= int(numNodes)
// TODO: default case?
			}
```

Reply:
The default case is covered with the last return in the function, we do not expect to have messages without type.

Reply:
In that case will we log the message? If it's not something expected, then we have kinda wrong .proto file version or something, and it'd be nice to log it as an error.